pipeline/lib/api.ml

```diff
@@ -111,11 +121,26 @@ let capture_metrics conninfo api_tokens =
           (status, response)
         with
         | Yojson.Json_error e | Yojson.Safe.Util.Type_error (e, _) | Failure e
-        ->
-          let response =
-            `Assoc [ ("success", `Bool false); ("error", `String e) ]
-          in
-          (`Bad_request, response)
+          ->
+            let response =
+              `Assoc [ ("success", `Bool false); ("error", `String e) ]
+            in
+            (`Bad_request, response)
+        | AuthError e ->
+            let response =
+              `Assoc [ ("success", `Bool false); ("error", `String e) ]
+            in
+            (`Unauthorized, response)
+        | PermissionError e ->
+            let response =
+              `Assoc [ ("success", `Bool false); ("error", `String e) ]
+            in
+            (`Forbidden, response)
+        | ServerError e ->
+            let response =
+              `Assoc [ ("success", `Bool false); ("error", `String e) ]
+            in
+            (`Internal_server_error, response)
```

Comment:
I don't know if we get "nice" error messages in the log if an unanticipated exception is raised? (and not catched by this `try`)
- user should get an "internal server error"
- we should get a dump of the error and backtrace in the log to investigate :)

It's very likely that ocurrent/cohttp already handle this! Otherwise look into adding a "catchall" `try ... with ... | error -> ...` and [Printexc](https://ocaml.org/api/Printexc.html) :)

Reply:
Yes, ocurrent/cohttp already handle this!